build.sh

```diff
@@ -435,13 +435,19 @@ build_CoreLib()
         exit 1
     fi
 
+    local __CoreLibILDir=$__BinDir/IL
+
     if [ $__SkipCrossgen == 1 ]; then
         echo "Skipping generating native image"
+
+        if [ $__CrossBuild == 1 ]; then
+            # Crossgen not performed, so treat the IL version as the final version
+            cp $__CoreLibILDir/System.Private.CoreLib.dll $__BinDir/System.Private.CoreLib.dll
```

Comment:
This is a somewhat scary change, as it is easy to forget that this SPC is not crossgened for one specific target. @RussKeldorph do you have an opinion.

Reply:
This behavior is same to the one before #24285, i.e. if `-skipcrossgen` was passed, SPC.dll was copied to bin dir in `MovePostBuildBinaries`

Reply:
Ah I see, I missed the skipcrossgen flag. Thanks, makes much more sense